specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/vnf.json

```diff
@@ -0,0 +1,431 @@
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.VnfManager/vnfs": {
+      "get": {
+        "tags": [
+          "VnfManagerVnf"
+        ],
+        "operationId": "VnfManagerVnfs_ListByResourceGroup",
+        "description": "Gets all the Vnf Manager Vnfs in a resource Group.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "$ref": "./vnfmanager.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./vnfmanager.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of Vnf Manager vnf resources.",
+            "schema": {
+              "$ref": "#/definitions/VirtualNetworkFunctionListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./vnfmanager.json#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List VnfManager vnf in resource group": {
+            "$ref": "./examples/VnfList.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "VnfManagerVnf": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/VnfManagerVnfFormat",
+          "description": "Vnf Manager vnf Properties."
+        },
+        "etag": {
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./vnfmanager.json#/definitions/Resource"
+        }
+      ],
+      "description": "Vnf Manager vnf resource response."
+    },
+    "VnfManagerVnfFormat": {
+      "description": "Vnf Manager Vnf response properties.",
+      "properties": {
+        "provisioningState": {
+          "readOnly": true,
+          "$ref": "./vnfmanager.json#/definitions/ProvisioningState",
+          "description": "The provisioning state of the Vnf Manager Vnf resource."
+        },
+        "mecDevice": {
+          "$ref": "#/definitions/MecDevice",
+          "description": "The reference to the Vnf Manager Device."
+        },
+        "sku": {
+          "type": "string",
+          "description": "The sku for the Vnf Manager Vnf."
+        },
+        "vendor": {
+          "type": "string",
+          "description": "The vendor for the Vnf Manager Vnf."
+        },
+        "vnfServiceKey": {
```

Comment:
>vnfServiceKey [](start = 9, length = 13)

if it is secret, annotate it with x-ms-secret:true

Reply:
I am getting the Schema Violation when I am using the property "x-ms-secret".
ERROR: Schema violation: Additional properties not allowed: x-ms-secret
- file:///C:/Users/neeleshk/azure-rest-api-specs/specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json:458:8 ($.definitions.DeviceRegistrationKey.properties.mecRegistrationKey)
I have not added this property but fixed other comments.